S32-list/duckmap.t

```diff
@@ -19,9 +19,11 @@ This test tests C<duckmap>.
     my $roles = ((1 but role {method foo {"a"}}), 2, ((3 but role {method foo {"b"}})));
     is-deeply $roles.duckmap({.foo}), ("a", 2, ("b")), "duckmap works with roles";
 
-    #?rakudo todo "RT #129363 duckmap doesn't preserve structure types"
-    is-deeply duckmap(-> Int $x { $x ~~ Int ?? $x !! Any }, [1, [2,3], 4]), [1, [2,3], 4],
-        'duckmap preserves structure types';
+    is-deeply my_duckmap({ .elems }, ["a", ["bb", "cc", "d"], []]), [1, 3, 0], "duckmap doesn't consume iterables for defined ops";
+
+    #RT #129363 duckmap doesn't preserve structure types"
+    is-deeply my_duckmap(-> Int $x { $x ~~ Int ?? $x !! Any }, [1, [2,3], 4]), [1, [2,3], 4], 'duckmap preserves structure types';
+    is-deeply my_duckmap(-> Int $x { $x ~~ Int ?? $x + 1 !! Any }, [1, [2,3], 4]), [2, [3,4], 5], 'duckmap preserves structure types';
```

Comment:
The `plan` on line 5 will need to be bumped to include the two extra added tests (so it should be `plan 7;`)